src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -98,20 +112,30 @@ static IList<XmlnsDefinitionAttribute> GetXmlnsDefinitionAttributes(Compilation
 					continue;
 				foreach (var attr in symbol.GetAttributes())
 				{
-					if (!SymbolEqualityComparer.Default.Equals(attr.AttributeClass, xmlnsDefinitonAttribute))
-						continue;
-					var xmlnsDef = new XmlnsDefinitionAttribute(attr.ConstructorArguments[0].Value as string, attr.ConstructorArguments[1].Value as string);
-					if (attr.NamedArguments.Length == 1 && attr.NamedArguments[0].Key == nameof(XmlnsDefinitionAttribute.AssemblyName))
-						xmlnsDef.AssemblyName = attr.NamedArguments[0].Value.Value as string;
-					else
-						xmlnsDef.AssemblyName = symbol.Name;
-					cache.Add(xmlnsDef);
+					if (SymbolEqualityComparer.Default.Equals(attr.AttributeClass, xmlnsDefinitonAttribute))
+					{
+						// [assembly: XmlnsDefinition]
+						var xmlnsDef = new XmlnsDefinitionAttribute(attr.ConstructorArguments[0].Value as string, attr.ConstructorArguments[1].Value as string);
+						if (attr.NamedArguments.Length == 1 && attr.NamedArguments[0].Key == nameof(XmlnsDefinitionAttribute.AssemblyName))
+							xmlnsDef.AssemblyName = attr.NamedArguments[0].Value.Value as string;
+						else
+							xmlnsDef.AssemblyName = symbol.Name;
+						xmlnsDefinitions.Add(xmlnsDef);
+					}
+					else if (SymbolEqualityComparer.Default.Equals(attr.AttributeClass, internalsVisibleToAttribute))
+					{
+						// [assembly: InternalsVisibleTo]
+						if (attr.ConstructorArguments[0].Value is string assemblyName && new AssemblyName(assemblyName).Name == compilation.Assembly.Identity.Name)
+						{
+							internalsVisible.Add(symbol);
+						}
+					}
```

Comment:
To avoid double iteration of all attributes on assemblies, I added this to a lager (new) cache object.